hordes-ui-mod.js

```diff
@@ -382,941 +392,1071 @@
+                modHelpers.selectChatTab(tabId);
+            }
+
+            // Returning tabId to all adding new tab to pass tab ID to `showChatTabConfigWindow`
+            return tabId;
+        },
+
+        // Selects chat tab [on click], updating client chat filters and custom chat filters
+        selectChatTab: tabId => {
+            // Remove selected class from everything, then add selected class to clicked tab
+            Array.from(document.querySelectorAll('[data-tab-id]')).forEach($tab => {
+                $tab.classList.remove('js-selected-tab');
+            });
+            const $tab = document.querySelector(`[data-tab-id="${tabId}"]`);
+            $tab.classList.add('js-selected-tab');
+
+            const tabFilters = state.chatTabs.find(tab => tab.id === tabId).filters;
+            // Simulating clicks on the filters to turn them on/off
+            const $filterButtons = Array.from(document.querySelectorAll('.channelselect small'));
+            Object.keys(tabFilters).forEach(filter => {
+                const $filterButton = $filterButtons.find($btn => $btn.textContent === filter);
+                const isCurrentlyFiltered = $filterButton.classList.contains('textgrey');
+
+                // If is currently filtered but filter for this tab is turned off, click it to turn filter off
+                if (isCurrentlyFiltered && !tabFilters[filter]) {
+                    $filterButton.click();
+                }
+                // If it is not currently filtered but filter for this tab is turned on, click it to turn filter on
+                if (!isCurrentlyFiltered && tabFilters[filter]) {
+                    $filterButton.click();
+                }
+            });
+
+            // Update state for our custom chat filters to match the tab's configuration, then filter chat for it
+            const isGMChatVisible = !tabFilters.GM;
+            modHelpers.setGMChatVisibility(isGMChatVisible);
+
+            // Update the selected tab in state
+            state.selectedChatTabId = tabId;
+            save();
+        },
+
+        // Updates state.chat.GM and the DOM to make text white/grey depending on if gm chat is visible/filtered
+        // Then filters chat and saves updated chat state
+        setGMChatVisibility: isGMChatVisible => {
+            const $chatGM = document.querySelector(`.js-chat-gm`);
+            state.chat.GM = isGMChatVisible;
+            $chatGM.classList.toggle('textgrey', !state.chat.GM);
+            modHelpers.filterAllChat();
+            save();
+        },
+
+        /** get char level */
+        getCurrentLvl: () => Number(document.querySelector('#ufplayer > div > div > div.progressBar.bgmana > .left').textContent.split('Lv. ')[1]),
+
+        /** get current xp */
+        getCurrentXp: () => Number(document.querySelector('#expbar > .bar > .progressBar > .left').textContent.split('/')[0].trim()),
+
+        /** get next level xp */
```

Comment:
Hope you don't mind the nitpickiness here - I'm cool with remove comments above methods that type out the method name.
Your method names alone are good here :)
`getNextLevelXp` is just as descriptive as the comment `get next level xp`, so we don't necessarily need to repeat ourselves.

I'm personally a fan of using comments to explain the `why` of the code, assuming there's a reason or need to explain that.

Feel free to remove (or if you feel necessary, expand on) the comments above any other methods that you think apply here.